zebra-chain/src/transaction/arbitrary.rs

```diff
@@ -768,6 +768,24 @@ impl Arbitrary for Transaction {
     type Strategy = BoxedStrategy<Self>;
 }
 
+impl Arbitrary for UnminedTx {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (any::<super::UnminedTxId>(), any::<Transaction>())
+            .prop_map(|(id, transaction)| Self {
+                id,
+                transaction: Arc::new(transaction.clone()),
+                size: transaction
+                    .zcash_serialized_size()
+                    .expect("all transactions have a size"),
+            })
+            .boxed()
```

Comment:
I think we always want the ID and transaction to match here.
(That's different from the derived implementation, but I think it's more correct.)

```suggestion
        any::<Transaction>().prop_map_into().boxed()
```